lib/js/components/TextGroup/TextGroup.vue

```diff
@@ -0,0 +1,296 @@
+<template>
+	<div
+		class="ds-textGroup"
+		:class="{
+			'-ds-small': size === TEXT_GROUP_SIZES.SMALL,
+
+			'-ds-hovered': state === TEXT_GROUP_STATES.HOVERED,
+			'-ds-loading': isLoading,
+			'-ds-disabled': state === TEXT_GROUP_STATES.DISABLED,
+
+			'-ds-interactive': isInteractive,
+			'-ds-selected': isSelected,
+
+			[loadingSizeClassName]: isLoading,
+
+			[colorClassName]: true,
+		}"
+	>
+		<div
+			v-if="eyebrowText !== null"
+			class="ds-textGroup__eyebrow"
+			:class="{
+				'-ds-ellipsis': eyebrowTextEllipsis,
+				'-ds-uppercase': isEyebrowTextUppercase,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="50%" height="100%" />
+			<span v-else>{{ eyebrowText }}</span>
+		</div>
+		<div
+			v-if="mainText !== null"
+			class="ds-textGroup__main"
+			:class="{
+				'-ds-ellipsis': mainTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else>{{ mainText }}</span>
+		</div>
+		<div
+			v-if="supportingText !== null"
+			class="ds-textGroup__supporting"
+			:class="{
+				'-ds-ellipsis': supportingTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else v-html="supportingText" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss">
+@import '../../../styles/settings/typography/tokens';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/spacings';
+
+.ds-textGroup {
+	$self: &;
+
+	display: flex;
+	flex-direction: column;
+	width: 100%;
+
+	&__eyebrow {
+		@include info-s-default-bold;
+
+		color: $color-neutral-text-weak;
+		margin-bottom: $space-5xs;
+		min-height: 12px;
+
+		&.-ds-uppercase {
+			@include info-s-extensive-bold-uppercase;
+		}
+	}
+
+	&__main {
+		@include text-m-compact-bold;
+
+		color: $color-neutral-text-strong;
+		min-height: 18px;
+	}
+
+	&__supporting {
+		@include text-s-compact-regular;
+
+		color: $color-neutral-text;
+		min-height: 16px;
+	}
+
+	&__eyebrow.-ds-ellipsis,
+	&__main.-ds-ellipsis,
+	&__supporting.-ds-ellipsis {
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+	}
+
+	&.-ds-small {
+		#{$self}__eyebrow {
+			@include info-xs-default-bold;
+
+			margin-bottom: 0;
+			min-height: 10px;
```

Comment:
```suggestion
			min-height: $typography-line-height-5xs;
```
it isn't perfect, as variable name doesn't match typography name, but it's little bit safer than pixel value

Reply:
There is an alternative way to achieve similar result without these `min-heights`. We could use `&nbsp;` or another not visible char to reserve space in this container. Not sure if it's cleaner, but definitely easier to maintain the same height during style changes